From eef0bedc46cad28fafa6a8470c16350cca0efab7 Mon Sep 17 00:00:00 2001
From: Roger Luethi <roger.luethi@bisdn.de>
Date: Tue, 13 Dec 2022 14:52:28 +0100
Subject: accton-csp7551-sfp: update for Linux 6.1

This commit updates the x86-64-accton-csp7551-sfp driver for Linux 6.1.

The coccinelle patch [1] failed to apply to this file. Therefore, the
fix was created manually.

[1] https://github.com/bisdn/meta-open-network-linux/blob/main/scripts/coccinelle/modules/6.1-i2c_remove_sig.cocci

Signed-off-by: Roger Luethi <roger.luethi@bisdn.de>
---
 .../files/x86-64-accton-csp7551-sfp.c                 | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/recipes-kernel/accton-csp7551-mods/files/x86-64-accton-csp7551-sfp.c b/recipes-kernel/accton-csp7551-mods/files/x86-64-accton-csp7551-sfp.c
index f38d231..f66f5a9 100755
--- a/recipes-kernel/accton-csp7551-mods/files/x86-64-accton-csp7551-sfp.c
+++ b/recipes-kernel/accton-csp7551-mods/files/x86-64-accton-csp7551-sfp.c
@@ -31,6 +31,7 @@
 #include <linux/sysfs.h>
 #include <linux/slab.h>
 #include <linux/delay.h>
+#include <linux/version.h>
 
 #define DRIVER_NAME     "csp7550_sfp" /* Platform dependent */
 
@@ -1953,6 +1954,12 @@ static int sfp_device_remove(struct i2c_client *client)
     return ret;
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,1,0)
+static void sfp_device_remove_6_1(struct i2c_client *client){
+    sfp_device_remove(client);
+}
+#endif
+
 /* Addresses scanned
  */
 static const unsigned short normal_i2c[] = { I2C_CLIENT_END };
@@ -1962,7 +1969,11 @@ static struct i2c_driver sfp_driver = {
         .name       = DRIVER_NAME,
     },
     .probe          = sfp_device_probe,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,1,0)
+    .remove         = sfp_device_remove_6_1,
+#else
     .remove         = sfp_device_remove,
+#endif
     .id_table       = sfp_device_id,
     .address_list   = normal_i2c,
 };
-- 
2.40.1

